README.md

```diff
@@ -433,6 +433,200 @@ to confirm that our debugging object is still displayed:
+  final promise result since we're only interested in side effects.
+
+> [!TIP]
+> 
+> Chaining promises with `then_` allows you to handle asynchronous code in a
+> structured way, similar to how you'd do it in JavaScript. We will see later on
+> a different way to handle asynchronous code, that is closer to `async/await`.
+
+### 6.4: Step 6 completion check
+
+Run your application to ensure that the data is being fetched correctly. Check
+your console to see the printed data. If everything is working, you should see
+the raw JSON data in your console.
+
+If everything works well, now it's a good time to clean up some of the previous
+prototyping code that we are not using anymore: `Feed.data`, `Feed.demoFeed` and
+the `Js.log(Feed.feed_of_json);` line at the end of `App.re`.
+
+## Step 7: Displaying data
+
+Now that we have successfully fetched the data, let’s move on to decoding it and
+displaying it in the UI.
+
+### 7.1: Decoding the fetched data
+
+We need to decode the raw data using our `feed_of_json` function. First of all,
+we are going to define a new variant type to specify the state of the UI:
+
+```reason
+type loadingStatus =
+  | Loading
+  | Loaded(result(Feed.feed, string));
+```
+
+This type uses a few things:
+- [Variant types](https://reasonml.github.io/docs/en/variant), similar to enums
+- Result type, which is a predefined variant with two options: `Ok` and `Error`
+- In this case, the "Ok" part is the `feed` type we defined in the previous
+  step, and the error type is just a string
+
+Now we can adapt the effect in `App.re` to use the new type. We will also be
+adding a `useState` hook to store the data loading state:
+
+
+```reason
+/* inside the App component `make` function */
+    let (data, setData) = React.useState(() => Loading);
+    React.useEffect0(() => {
+      Js.Promise.(
+        Fetch.fetch("https://gh-feed.vercel.app/api?user=jchavarri&page=1")
+        |> then_(Fetch.Response.text)
+        |> then_(text =>
+             {
+               let data =
+                 try(Ok(text |> Json.parseOrRaise |> Feed.feed_of_json)) {
+                 | Json.Decode.DecodeError(msg) =>
+                   Js.Console.error(msg);
+                   Error("Failed to decode: " ++ msg);
+                 };
+               setData(_ => Loaded(data));
+             }
+             |> resolve
+           )
+      )
+      |> ignore;
+      None;
+    });
+```
+
+### 7.2: Rendering the data in the UI
+
+With the data decoded, it’s time to display it in the UI. Update your React
+component to process the result and render the feed:
+
+```reason
+  {switch (data) {
+    | Loading => <div> {React.string("Loading...")} </div>
+    | Loaded(Error(msg)) => <div> {React.string(msg)} </div>
+    | Loaded(Ok(feed)) =>
+      <div>
+        <h1> {React.string("GitHub Feed")} </h1>
+        <ul>
+          {feed.entries
+          |> Array.map((entry: Feed.entry) =>
+                <li key={string_of_float(entry.updated)}>
```

Comment:
hot take: `string_of_float` is rather confusing for people

Reply:
How would you write this otherwise?

Reply:
`Int.to_string(entry.id)`

Reply:
Maybe mention that `Js.Int.toString` is also available